forem/forem
# Profile generalization for the onboarding workflow

## NOTE

**There's a [pre deployment task](https://github.com/forem/forem/pull/11704) that will need to be run for existing Forems before this PR can be merged.**

## What type of PR is this? (check all applicable)


## Description
We currently have Profile Generalization implemented as a feature flag. In order to bring this feature to completion, we need to implement the profile generalization across multiple areas. This includes the profile creation in the onboarding process which is currently hardcoded.

A related Slack thread for anyone with access that's interested in some the decisions: https://dev-internal.slack.com/archives/C0186BE65TN/p1606752839018000

## Related Tickets & Documents
https://github.com/forem/forem/issues/11529

## QA Instructions, Screenshots, Recordings

- Enable the profile generalization feature flag. You can follow the instructions [here](https://github.com/forem/forem/pull/11149) or just open up a rails console and run `Flipper.enable(:profile_admin)`
- Go http://localhost:3000/admin/profile_fields
- Configure the Profile fields and tick `show in onboarding` for some of the fields. 
 - The fields that you've ticked should now show up in "Step 2: Build your profile" of the onboarding section http://localhost:3000/onboarding.
- Fill out the fields in this step and make sure they persist by checking the user profile page http://localhost:3000/settings/profile or just looking at the profile information in the console.

If you're not familiar with the feature, you may stumble on replicating this so if anyone needs more verbose instructions, please feel free to ping me and I can make a video or provide more in depth instructions and add it to the PR.

### UI accessibility concerns?

_If your PR includes UI changes, please replace this line with details on how accessibility is impacted and tested. For more info, check out the [Forem Accessibility Docs](https://docs.forem.com/frontend/accessibility)._

## Added tests?

      have not been included_

## Added to documentation?


## [optional] Are there any **pre deployment** tasks we need to perform?
https://github.com/forem/forem/pull/11704 needs to first be merged and run.

## [optional] What gif best describes this PR or how it makes you feel?

![alt_text](https://media.giphy.com/media/l0NwNrl4BtDD7JCx2/giphy.gif)
